### PR TITLE
If restarting fails, the server must be kept responsive

### PR DESCRIPTION
**Pull request by mcollina**
Signed-off-by: Matteo Collina <hello@matteocollina.com>

Fixes https://github.com/platformatic/platformatic/issues/85

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -38,25 +40,37 @@ async function start (opts) {
...
-    old.server.removeListener('request', oldHandler)
-    newApp.server.on('request', newApp.server.handler)
-    for (const listener of clientErrorListeners) {
-      old.server.removeListener('clientError', listener)
-    }
-    res.app = newApp
+    try {
+      await newApp.ready()
+      old.server.removeListener('request', oldHandler)
+      old.server.removeListener('request', oldHandler)
```
Comment (ivan-tymoshenko) [inline index.js:55]: This is not changed, but why there is a duplicate `removeListener` call?

File: index.js [hunk 2]
```diff
@@ -38,25 +40,37 @@ async function start (opts) {
...
+      newApp.server.on('request', handler)
+      for (const listener of clientErrorListeners) {
+        old.server.removeListener('clientError', listener)
+      }
+      res.app = newApp
 
-    await old.close()
+      await old.close()
+    } catch (err) {
+      const listenersNow = newApp.server.listeners('clientError')
```
Comment (ivan-tymoshenko) [inline index.js:64]: I don't understand why we need to update the listener's list. Why can't we use `clientErrorListeners`?

Comment (mcollina) [reply index.js:55]: I've added a comment explaining this.

**Review by ivan-tymoshenko — APPROVED** [review]
LGTM

**Pull request merged**